lib/logstash/inputs/elasticsearch.rb

```diff
@@ -198,14 +206,25 @@ def stop
   def do_run(output_queue)
     # get first wave of data
     r = @client.search(@options)
-
-    r['hits']['hits'].each { |hit| push_hit(hit, output_queue) }
-    has_hits = r['hits']['hits'].any?
-
-    while has_hits && !stop?
-      r = process_next_scroll(output_queue, r['_scroll_id'])
-      has_hits = r['has_hits']
+    case @response_type
+      when 'hits'
+        r['hits']['hits'].each { |hit| push_hit(hit, output_queue) }
+        has_hits = r['hits']['hits'].any?
+
+        while has_hits && !stop?
+          r = process_next_scroll(output_queue, r['_scroll_id'])
+          has_hits = r['has_hits']
+        end
+      when 'aggregations'
+        aggs = r['aggregations']
+        event = LogStash::Event.new(aggs)
+
+        decorate(event)
+        output_queue << event
```

Comment:
Maybe it's cleaner to put these 4 lines in a `push_aggregation` method, to be consistent with `push_hit` method ?